src/go/configgenerator/route_generator.go

```diff
@@ -167,7 +167,7 @@ func MakePathRewriteConfig(method *configinfo.MethodInfo, httpRule *httppattern.
 		}
 
 		if uriTemplate := httpRule.UriTemplate; uriTemplate != nil && len(uriTemplate.Variables) > 0 {
-			constPath.UrlTemplate = uriTemplate.String()
+			constPath.UrlTemplate = uriTemplate.ExactMatchString(false)
```

Comment:
NIT: add `acceptTrailingBackslash` comment before argument for readability.

Reply:
I assume you're talking about this in C++? http://clang.llvm.org/extra/clang-tidy/checks/bugprone-argument-comment.html

This is not only for readability, it's actually checked by `clang-tidy`.

I suggest not doing this in golang:
- It's not part of the golang style guide
- No golang linter will check to ensure they're correct (especially after refactoring, they can go out of sync)